mokapot/brew.py

```diff
@@ -144,38 +192,91 @@ def brew(psms, model=None, test_fdr=0.01, folds=3, max_workers=1, rng=None):
     # If we reset, just use the original model on all the folds:
     if reset:
         scores = [
-            p._calibrate_scores(model.predict(p), test_fdr) for p in psms
+            _psms.calibrate_scores(
+                _predict_with_ensemble(
+                    psms=_psms,
+                    models=[model],
+                    max_workers=max_workers,
+                ),
+                test_fdr,
+            )
+            for _psms in psms
         ]
 
     # If we don't reset, assign scores to each fold:
     elif all([m.is_trained for m in models]):
-        scores = [
-            _predict(p, i, models, test_fdr) for p, i in zip(psms, test_idx)
-        ]
-
+        if ensemble:
```

Comment:
I agree and just spent some time to try to factor it out. I would wish to do the following:

- creating a new submodule `mokapot.scoring` and move the whole section into a function called get_scores(...)
- the `_predict` and `_predict_with_ensemble` should also live in `mokapot.scoring`

Naively doing so would create cyclic dependencies between `scoring` and `brew` because both rely on `_create_psms`. To resolve the cyclic dependency, we would need to factor all data preprocessing functions into their own submodule as well. 

I think this would be definitely desirable and would clean up `brew.py` a lot but I would prefer this in a separate smaller issue / PR after this one, if that would be ok.

Reply:
The brew function is currently nearly 270 lines long, so IMHO this would be very desirable, too.

Reply:
https://github.com/wfondrie/mokapot/pull/119#issuecomment-2231943971 <- Added there to the series of things that should be done but not now